qtconsole/console_widget.py

```diff
@@ -396,6 +399,7 @@ def eventFilter(self, obj, event):
             text widgets.
         """
         etype = event.type()
+        self._trigger_is_complete_callback()
```

Comment:
Aha, I see. So if the uses presses any other key before the `is_complete_reply` arrives, we assume it's incomplete and let them carry on typing?

That sounds reasonable, but have you tested it with a kernel that doesn't implement `is_complete`? I'm not sure what to recommend to test with, however; I thought IRkernel didn't implement it, but when I went to check, it does. :-)